docs/source/architecture.rst

```diff
@@ -1,17 +1,134 @@
+  tolerance, view changes, sequence numbers, and the information stored
+  by each node
+
+* :ref:`Consensus messages <consensus-messages-label>`: Explains consensus
+  message structures and message types
+
+* :ref:`Sawtooth PBFT operation <pbft-operation-label>`: Shows how the
+  algorithm handles initialization, normal mode (block processing), and view
+  changes
+
+.. _network-overview-label:
+
+Network Overview
+================
+
+Fault Tolerance
+---------------
+
+A PBFT network consists of nodes that are ordered from 0 to `n-1`, where
+`n` is the total number of nodes in the network. The
+:doc:`on-chain setting <on-chain-settings>` ``sawtooth.consensus.pbft.peers``
+lists all nodes and determines the node order.
+
+The PBFT algorithm guarantees network `safety
+<https://en.wikipedia.org/wiki/Liveness#Liveness_and_safety>`__
+as long as the number of faulty nodes remains below the required percentage.
+The maximum number of faulty nodes that the network can tolerate is determined
+by the formula :math:`f = \frac{n - 1}{3}`. In other words, no more than one
+third of the nodes (rounded down) can be unreachable or dishonest at any given
+time.
+
+For example, a four-node network can tolerate one faulty node. (PBFT requires a
+minimum of four nodes in order to maintain Byzantine fault tolerance.)
+Increasing the size of the network reduces the likelihood that all
+:math:`\frac{n - 1}{3}` nodes would be faulty at the same time.
+
+View Changes: Choosing a New Primary
+------------------------------------
+
+A `view` is the period of time that a given node is the primary, so a `view
+change` means switching to a different primary node. The next primary is
+selected in a round-robin (circular) fashion, according to the order of nodes
+listed in the :doc:`on-chain setting <on-chain-settings>`
+``sawtooth.consensus.pbft.peers``.
+
+In a four-node network, for example, the first node (node 0) is the primary at
+view 0, the second node (node 1) is the primary at view 1, and so on.  When the
+network gets to view 4, it will "wrap back around" so that node 0 is the primary
+again.
+
+The algorithm uses the formula `p = v mod n` to determine the next
+primary. In this formula, `p` is the primary, `v` is the view number, and `n` is
+the total number of nodes in the network. For example, if a four-node network is
+at view 7, the formula `7 mod 4` determines that node 3 (the fourth node) is the
+primary.
+
+The Sawtooth PBFT algorithm changes the primary at regular intervals, as well as
+when the secondary nodes determine that the current primary is faulty.
+See :ref:`view-changing-mode-label` for a description of this process.
+
+Sequence Numbers
+----------------
+
+With PBFT, each new block has a `sequence number`. Sawtooth PBFT uses the block
+number as the sequence number.
+
+For example, a node that is on sequence 10 is evaluating block 10.
+Note that this is actually the 11th block, because both sequence and block
```

Comment:
I'd remove this sentence; we never really refer to a block as the "11th" block, we'd just refer to it as block 10.